### PR TITLE
Re-init changes + add refresh option to refresh config file

### PR DESCRIPTION
**Pull request by IgorTodorovskiIBM**
* re-init: this PR fixes that by checking the value of ZOPEN_ROOTFS
* Adds a refresh option to refresh the config file
* Refactors zopen-init code into a few functions
* Fixes for update-cacert

### CODE REVIEW

**Review by MikeFultonDev — APPROVED** [review]
LGTM

**Comment by MikeFultonDev** [comment]
good to see our testing catching issues!

**Comment by IgorTodorovskiIBM** [comment]
Tests are now passing

**Review by DevonianTeuchter — APPROVED** [review]
LGTM

File: bin/zopen-init [hunk 1]
```diff
@@ -131,283 +138,328 @@ while [ $# -gt 0 ]; do
...
+          reinit=y
+        else
+          printWarning "- Existing configuration file found; re-initializing might lose access to previously installed packages. Continue (y/n)?"
+          reinit=$(getInput)
+        fi
+      fi
+      if ${reinitExisting} || {
+        [ -n "${reinit}" ] && [ "y" = "${reinit}" ]
+      }; then
+        printInfo "- Re-initializing; removing existing zopen-config"
```
Comment (MikeFultonDev) [inline bin/zopen-init:226]: should we say 'removing' or 'updating'? This might scare people.

File: bin/zopen-init [hunk 2]
```diff
@@ -131,283 +138,328 @@ while [ $# -gt 0 ]; do
...
+
+  printInfo "- Creating symbolic path for prod redirect files"
+  [ -e "${rootfs}/usr/share/zopen/boot" ] || ln -s "${rootfs}/boot" "${rootfs}/usr/share/zopen/boot"
+  [ -e "${rootfs}/etc/zopen" ] || mkdir -p "${rootfs}/etc/zopen"
+  echo "${zopen_pkginstall}" > "${rootfs}/etc/zopen/fstype"
+
+  [ -e "${rootfs}/${zopen_pkginstall}" ] || mkdir -p "${rootfs}/${zopen_pkginstall}"
+  [ -e "${rootfs}/usr/share/zopen/prod" ] || ln -s "${rootfs}/${zopen_pkginstall}" "${rootfs}/usr/share/zopen/prod"
+
+  if [ "${ZOPEN_PKGINSTALL}" = "${rootfs}" ]; then
```
Comment (MikeFultonDev) [inline bin/zopen-init:269]: if it isn't recommended why not just error out?

File: bin/zopen-init [hunk 3]
```diff
@@ -131,283 +138,328 @@ while [ $# -gt 0 ]; do
...
-  printWarning "Package install location is the zopen root location; this is not recommended. Continue (y/n)"
-  pkgoverlaysrootfs=$(getInput)
-  if [ -n "${pkgoverlaysrootfs}" ] && [ "y" = "${pkgoverlaysrootfs}" ]; then
-    printInfo "- Continuing. Packages will be installed into the zopen root location: ${rootfs}"
-  else
-    printWarning "- Stopping. Existing configuration will be preserved"
-    exit 8
-  fi
-fi
+  printInfo "- Creating path for certificate lookups"
```
Comment (MikeFultonDev) [inline bin/zopen-init:279]: do you want to say where?

Comment (MikeFultonDev) [reply bin/zopen-init:279]: if they will never need to know and it's internal, it's ok to not say

File: bin/zopen-init [hunk 4]
```diff
@@ -131,283 +138,328 @@ while [ $# -gt 0 ]; do
...
-printInfo "- Creating path for certificate lookups"
-[ -e "${rootfs}/${ZOPEN_CA_DIR}" ] || mkdir -p "${rootfs}/${ZOPEN_CA_DIR}"
+generateConfig()
+{
+  
+  writeConfigFile "${configFile}" "${rootfs}" "${zopen_pkginstall}" "${ZOPEN_CA_DIR}/${certFileName}"
+  printInfo "- Created config in ${configFile}"
+  if ${appendToProfile}; then
+    if ! grep -q ". ${rootfs}/etc/zopen-config" "${HOME}/.profile"; then
+      printVerbose "Line not present (grep returned non-0)"
```
Comment (MikeFultonDev) [inline bin/zopen-init:293]: what line isn't present - not sure this would be useful for someone

File: bin/zopen-init [hunk 5]
```diff
@@ -131,283 +138,328 @@ while [ $# -gt 0 ]; do
...
+    fi
+  fi
+}
 
-writeConfigFile "${configFile}" "${rootfs}" "${zopen_pkginstall}" "${ZOPEN_CA}"
-printInfo "- Created config in ${configFile}"
+setDefaults()
+{
+  # Generate any system values that can be user changed
+  echo "15" > "${rootfs}/etc/zopen/rm_fileprocs"
```
Comment (MikeFultonDev) [inline bin/zopen-init:304]: can you add a comment for the magic number?

File: bin/zopen-init [hunk 6]
```diff
@@ -131,283 +138,328 @@ while [ $# -gt 0 ]; do
...
-  if [ -r "${ZOPEN_CA}" ]; then
-    printInfo "- Found existing setting for ZOPEN_CA '${ZOPEN_CA}'; reusing certificate"
-    if [ "${ZOPEN_CA}" != "${ZOPEN_ROOTFS}/${ZOPEN_CA_DIR}/${certFileName}" ]; then # avoid cp error
-      cp -f "${ZOPEN_CA}" "${ZOPEN_ROOTFS}/${ZOPEN_CA_DIR}"
-    fi
-    cacertFound=true
+installPrereqs()
+{
+  printInfo "- Check for shipped curl pax"
+  curlpax="packages/curl-8.4.0.20231011_155722.zos.pax.Z"
```
Comment (MikeFultonDev) [inline bin/zopen-init:319]: should this be some sort of env var? presumably every time we get a CVE in curl we will need to update this?

File: bin/zopen-init [hunk 7]
```diff
@@ -131,283 +138,328 @@ while [ $# -gt 0 ]; do
...
-    printInfo "- Found default certificate file '${certFileName}'; copying to zopen file system"
-    cp -f "${cacertlcn}/${certFileName}" "${ZOPEN_ROOTFS}/${ZOPEN_CA_DIR}"
-    cacertFound=true
+  printVerbose "- Expanding curl to bootstrap dir..."
+  cachedir="${rootfs}/var/cache/zopen"
+  basecurlpax=$(basename ${curlpax})
+  [ -e "${cachedir}/${basecurlpax}" ] || printError "Could not locate bootstrap curl pax."
+  paxrc=$(pax -rf "${cachedir}/${basecurlpax}" -s##${rootfs}/boot/#)
+
+  printInfo "- Check for shipped jq pax"
```
Comment (MikeFultonDev) [inline bin/zopen-init:335]: should this be verbose instead of info?

File: bin/zopen-init [hunk 8]
```diff
@@ -131,283 +138,328 @@ while [ $# -gt 0 ]; do
...
+  . "${configFile}"
+
+  printInfo "Updating CA Certificate"
+  export ZOPEN_CA="${ZOPEN_ROOTFS}/${ZOPEN_CA_DIR}/${certFileName}"
+  caCertDir="${ZOPEN_ROOTFS}/${ZOPEN_CA_DIR}"
+  mkdir -p "${caCertDir}"
+  runAndLog "${MYDIR}/zopen-update-cacert -f "${caCertDir}" -v"
+  [ $? -ne 0 ] && exit $?
+
+  printInfo "- Using bootstrapped curl and jq to install updated versions (if available)"
```
Comment (MikeFultonDev) [inline bin/zopen-init:381]: i think most infos can probably be verbose?

**Review by MikeFultonDev — COMMENTED** [review]
Mostly minor comments other than the lack of the prompt at the start being removed

**Review by MikeFultonDev — COMMENTED** [review]
nice update for curl. LGTM

**Review by MikeFultonDev — APPROVED** [review]
LGTM

**Pull request merged**